src/ContextMenu.js

```diff
@@ -89,10 +95,10 @@ export default class ContextMenu extends AbstractMenu {
     registerHandlers = () => {
         document.addEventListener('mousedown', this.handleOutsideClick);
         document.addEventListener('touchstart', this.handleOutsideClick);
-        document.addEventListener('scroll', this.handleHide);
-        document.addEventListener('contextmenu', this.handleHide);
+        if (!this.props.preventHideOnScroll) document.addEventListener('scroll', this.handleHide);
+        if (!this.props.preventHideOnContextMenu) document.addEventListener('contextmenu', this.handleHide);
         document.addEventListener('keydown', this.handleKeyNavigation);
-        window.addEventListener('resize', this.handleHide);
+        if (!this.props.preventHideOnResize) window.addEventListener('resize', this.handleHide);
```

Comment:
Probably add the same in `unregisterHandlers`?

Reply:
I'm worried about a possibility that any of this props will change from false to true after `registerHandlers` but before unregister. It seems safer to unsubscribe without condition. What do you think?

Reply:
Yeah that sounds legit